### DISCUSSION TITLE
No Roles Have App Access After Migration from v8

### DISCUSSION THREAD

**Discussion by salohcin714** [directus/directus#9620] [Archive]
Hello,

We were able to successfully use the migration script to get all of our data over to the new version. We were then able to get in and verify that schemas were somewhat all correct. However, before we could reconfigure permissions, the app started telling us that we didn't have app access. We are using administrator roles and have even created a new role through the API that has app access and we are still having issues. We believe it may have to do with the old v8 permissions.

We are running MySQL 8

**Comment by joselcvarela** [comment]
Hello @salohcin714 
What about `directus_users`? Do each row have the proper `role`?
This field should not be empty.

**Reply by salohcin714** [reply]
When I pull users from the API, I see all but one user having a role. Is the one having a null role breaking it all?

edit: That user is not one that we are using to login with

**Reply by salohcin714** [reply]
@joselcvarela I updated the user using the API and can see that now all users have roles assigned. Should I be checking the actual database table as well? I would assume that the API is pulling directly from it.

**Reply by salohcin714** [reply]
I checked in database and all records have an assigned role

**Reply by joselcvarela** [reply]
Try to set `NODE_ENV=development` and then check Network tab on developer tools. 
You should have a request with red colour. Check what it tells about the error you are getting.

**Reply by salohcin714** [reply]
@joselcvarela I've tried setting that in both .env and in the server environment variables and Directus still starts in production mode... As soon as I figure out how to get it in development mode, I will post results

**Reply by salohcin714** [reply]
@joselcvarela I am able to see the following if I have Directus print out `process.env`:

```
{
  ...
  "SERVE_APP": true,
  "NODE_ENV": "development",
  "DB_CLIENT": "mysql",
  ...
}
```
However, in the console, Directus still says `Environment: production` in the console. I am not seeing any failed network requests. I also went ahead and upgraded from RC99 to 9.0.0 and ran all migrations. I am still seeing No App Access

**Reply by joselcvarela** [reply]
If you do not have any red request, then we can forget the `NODE_ENV` thing.
Can you check if the role (`directus_roles`) has `app_access` equals true on your database? 
In order to users have access to the app, this column should be set to true.
If you want to have privileged access, like editing data model, you can also set `admin_access` to true.

**Reply by salohcin714** [reply]
@joselcvarela This is what I see when I pull `directus_roles`

```
+--------------+------------+
| admin_access | app_access |
+--------------+------------+
|            0 |          1 |
|            1 |          1 |
|            1 |          1 |
|            0 |          1 |
|            1 |          1 |
|            0 |          1 |
+--------------+------------+
```

**Reply by joselcvarela** [reply]
Unfortunately, I am not seeing what is causing this issue.
My last suggestion is to run Directus locally and use `console.log` and `debugger` to find out where the problem is: https://docs.directus.io/contributing/running-locally/#running-locally

Also you could give a shot on our discord: https://directus.chat
